src/click/decorators.py

```diff
@@ -342,19 +342,16 @@ def version_option(
         frame = inspect.currentframe()
         assert frame is not None
         assert frame.f_back is not None
-        f_globals = frame.f_back.f_globals if frame is not None else None
+        assert frame.f_back.f_globals is not None
```

Comment:
I think the report in the issue means that all three asserts here are not needed.

Reply:
Well, there are two ways to go: 
 - Using the `asserts` and removing all null checks (as we'd be certain there are no null values)
 - removing the assertions and check for null values

Reply:
I collapsed everything as you mentioned below. This is surrounded by a try block that checks for attribute error. tThis would be thrown when accessing attributes of a NoneType object.

```
        try:
            f_globals = inspect.currentframe().frame.f_back.f_globals
        except AttributeError:
            assert False
```